pkg/gardenlet/controller/shoot/shoot/reconciler_migrate.go

```diff
@@ -306,23 +335,43 @@ func (r *Reconciler) runMigrateShootFlow(ctx context.Context, o *operation.Opera
 			Dependencies: flow.NewTaskIDs(waitUntilExtensionsAfterKubeAPIServerMigrated),
 		})
 		migrateInfrastructure = g.Add(flow.Task{
-			Name:         "Migrating shoot infrastructure",
-			Fn:           botanist.Shoot.Components.Extensions.Infrastructure.Migrate,
+			Name: "Migrating shoot infrastructure",
+			Fn: flow.TaskFn(func(ctx context.Context) error {
+				if isWorkerless {
+					return nil
+				}
+				return botanist.Shoot.Components.Extensions.Infrastructure.Migrate(ctx)
+			}),
 			Dependencies: flow.NewTaskIDs(waitUntilKubeAPIServerDeleted),
 		})
 		waitUntilInfrastructureMigrated = g.Add(flow.Task{
-			Name:         "Waiting until shoot infrastructure has been migrated",
-			Fn:           botanist.Shoot.Components.Extensions.Infrastructure.WaitMigrate,
+			Name: "Waiting until shoot infrastructure has been migrated",
+			Fn: flow.TaskFn(func(ctx context.Context) error {
+				if isWorkerless {
+					return nil
+				}
+				return botanist.Shoot.Components.Extensions.Infrastructure.WaitMigrate(ctx)
+			}),
 			Dependencies: flow.NewTaskIDs(migrateInfrastructure),
 		})
 		deleteInfrastructure = g.Add(flow.Task{
-			Name:         "Deleting shoot infrastructure",
-			Fn:           botanist.Shoot.Components.Extensions.Infrastructure.Destroy,
+			Name: "Deleting shoot infrastructure",
+			Fn: flow.TaskFn(func(ctx context.Context) error {
+				if isWorkerless {
+					return nil
+				}
+				return botanist.Shoot.Components.Extensions.Infrastructure.Destroy(ctx)
+			}),
 			Dependencies: flow.NewTaskIDs(waitUntilInfrastructureMigrated),
 		})
 		waitUntilInfrastructureDeleted = g.Add(flow.Task{
-			Name:         "Waiting until shoot infrastructure has been deleted",
-			Fn:           botanist.Shoot.Components.Extensions.Infrastructure.WaitCleanup,
+			Name: "Waiting until shoot infrastructure has been deleted",
+			Fn: flow.TaskFn(func(ctx context.Context) error {
+				if isWorkerless {
+					return nil
+				}
+				return botanist.Shoot.Components.Extensions.Infrastructure.WaitCleanup(ctx)
+			}),
```

Comment:
Is `deployGardenerAccess`, `initializeShootClients` steps required for workerless?

Reply:
Yes.